lib/fastlane/plugin/wpmreleasetoolkit/helper/git_helper.rb

```diff
@@ -173,8 +173,16 @@ def self.ensure_on_branch!(branch_name)
       #
       # @return [Bool] True if the given path is ignored, false otherwise.
       def self.is_ignored?(path:)
-        Actions.sh('git', 'check-ignore', path) { |status, _, _| status.success? }
+        Actions.sh('git', 'check-ignore', path) do |status, output, _|
+          return true if output.strip == NOT_A_REPO_ERROR
+          return true if output.include? OUTSIDE_OF_REPO_ERROR
+
+          status.success?
+        end
       end
+
+      OUTSIDE_OF_REPO_ERROR = 'is outside repository at'.freeze
+      NOT_A_REPO_ERROR = 'fatal: not a git repository (or any of the parent directories): .git'.freeze
```

Comment:
@jkmassel @AliSoftware what do you think of this approach? 

I'm not particularly happy about checking error message strings, but the exit status of `git check-ignore` is not refined enough (see `git check-ignore --help`):

![image](https://user-images.githubusercontent.com/1218433/117249280-a9a4d480-ae84-11eb-8c5f-32985cb5a265.png)

I thought about doing some checks on the path here manually, but it felt like reinventing the wheel.

Reply:
I'm not a fan of matching error message either.

What about testing if this is a git repo separately, before invoking `check-ignore`?

I'm especially thinking about something like `git -C #{path} rev-parse --show-toplevel` 🤔

Reply:
Nice suggestion. I had a note to look into `rev-parse`, but started with the apparently simpler solution from this PR.

What attracts me about your idea is that it will allow us to prevent decrypting into _any_ repository, which is a nice benefit to have. Just in case someone misconfigures the tool and decrypts a file into a local repo on their machine, then end up pushing and leaking from there.

🤔 This also makes me think we might want to raise if a user is trying to decrypt to a non generic path, namely `/Users/$USERNAME`.

Reply:
Hey @AliSoftware, I'm afraid the `rev-parse` option is a dead end for what we're trying to achieve here.

I implemented it `a7a3961`, but discovered an issue: `-C $PATH` expects the path to exist already. This is a problem for the use case in which we want to decrypt new secrets to a path that doesn't exist yet.

I added a dedicated example, `af5fe9e`, and extra logs in the `rev-parse` implementation, `a265e09` to showcase this.

[Here's the CI build failing](https://app.circleci.com/pipelines/github/wordpress-mobile/release-toolkit/765/workflows/a7bd7906-9c57-410b-a4f1-2bca7ec750ac/jobs/739/steps). When used with `-C` and a path that doesn't exist yet, Git errors with:

```
fatal: cannot change to '$PATH': No such file or directory
```

That's disappointing 😞 but I can't blame Git for working that way...

I reverted to the original error output diff implementation mainly to have CI show that it works, but I bumped into a different issue in CI which I'll need to look into next. 🙃

Reply:
I see. Unfortunate but makes sense.

Then other option I see is to File.dirname recursively until File.exist? is true then finally use that for rev-parse 🧐

Reply:
I'll give it a shot and report 🧐